profilewatch/profilewatch.py

```diff
@@ -0,0 +1,155 @@
+import re
+from datetime import datetime
+from typing import Optional
+
+import discord
+import discord.utils
+from redbot.core import Config, checks, commands
+from redbot.core.bot import Red
+from redbot.core.utils.chat_formatting import escape, pagify
+from redbot.core.utils.menus import menu, prev_page, close_menu, next_page
+
+
+class ProfileWatchCog(commands.Cog):
+    """ProfileWatch Cog"""
+
+    def __init__(self, bot: Red):
+        self.bot = bot
+        self.config = Config.get_conf(self, identifier=128986274420752384001)
+
+        default_guild_config = {
+            "logchannel": "",  # Channel to send alerts to
+            "rules": {
+                "spammer1": {  # Name of rule
+                    "pattern": "^portalBlock$",  # Regex pattern to match against
+                    "check_nick": False,  # Whether the user's nickname should be checked too
+                    "alert_level": "HIGH",  # Severity of alerts (use: HIGH or LOW)
+                    "reason": "Impostor :sus:"  # Reason for the match, used to add context to alerts
+                }
+            }
+        }
+
+        self.config.register_guild(**default_guild_config)
+
+    @commands.Cog.listener()
+    async def on_member_join(self, member: discord.Member):
+
+        matcher_list = await self.config.guild(member.guild).rules()
+
+        for ruleName, rule in matcher_list.items():
+            hits = len(re.findall(rule['pattern'], member.name))
+            if rule['check_nick'] and member.nick:
+                hits += len(re.findall(rule['pattern'], member.nick))
+            if hits > 0:
+
+                # Credit: Taken from report Cog
+                log_id = await self.config.guild(member.guild).logchannel()
+                log = None
+                if log_id:
+                    log = member.guild.get_channel(log_id)
+                if not log:
+                    # Failed to get the channel
+                    return
+
+                data = self.make_alert_embed(member, ruleName, rule)
+
+                mod_pings = ""
+                # Alert level logic added
+                if rule['alert_level'] == "HIGH":
+                    mod_pings = " ".join(
+                        [i.mention for i in log.members if not i.bot and str(i.status) in ["online", "idle"]])
+                    if not mod_pings:  # If no online/idle mods
+                        mod_pings = " ".join([i.mention for i in log.members if not i.bot])
+
+                await log.send(content=mod_pings, embed=data)
+                # End credit
+
+    # Command groups
+
+    @checks.admin()
+    @commands.group(name="profilewatch", aliases=["pw"], pass_context=True)
+    async def _profilewatch(self, ctx):
+        """Monitor for flagged member name formats"""
+
+    # Commands
+
+    @_profilewatch.command(name="add")
+    async def _add(self, ctx, name: str = "", regex: str = "", alert_level: str = "",
```

Comment:
I believe discord.py has a built-in `bool` converter. Might be best to use `Optional[bool]` here rather than just typing everything as `str` with a default 🤔
Or do `bool` with a default.